### PR TITLE
Replace deprecated constants

### LINKED ISSUE #156: Problem with NSRightTextAlignment (and NSCenterTextAlignment)
Apple seems to have switch the values of those two constants in macOS12 and pyobjc is not correctly handling that. I reported it there: https://github.com/ronaldoussoren/pyobjc/issues/401

The easiest would be to switch to the newer API and use `NSTextAlignmentRight` (and `NSTextAlignmentCenter`). Those seem to work fine.

### PR DESCRIPTION
**Pull request by schriftgestalt**
fixes #156

### CODE REVIEW

File: Documentation/source/conf.py [hunk 1]
```diff
@@ -236,7 +236,7 @@ def __getattr__(cls, name):
...
...'NSXMLDocumentXInclude', 'NSXMLDocumentXMLKind', 'NSXMLElement', 'NSXMLElementDeclarationAnyKind', 'NSXMLElementDeclarationContent', 'NSXMLElementDeclarationElementKind', 'NSXMLElementDeclarationEmptyKind', 'NSXMLElementDeclarationKind', 'NSXMLElementDeclarationMixedKind', 'NSXMLElementDeclarationUndefinedKind', 'NSXMLElementKind', 'NSXMLEntityDeclarationKind', 'NSXMLEntityGeneralKind', 'NSXMLEntityParameterKind', 'NSXMLEntityParsedKind', 'NSXMLEntityPredefined', 'NSXMLEntityUnparsedKind', 'NSXMLFidelityElement', 'NSXMLFidelityNode', 'NSXMLInvalidKind', 'NSXMLNSArrayTransformerName', 'NSXMLNSDataTransformerName', 'NSXMLNSDateTransformerName', 'NSXMLNSNumberTransformerName', 'NSXMLNSURLTransformerName', 'NSXMLNamedFidelityNode', 'NSXMLNamedNode', 'NSXMLNamespaceKind', 'NSXMLNode', 'NSXMLNodeCompactEmptyElement', 'NSXMLNodeExpandEmptyElement', 'NSXMLNodeIsCDATA', 'NSXMLNodeOptionsNone', 'NSXMLNodePreserveAll', 'NSXMLNodePreserveAttributeOrder', 'NSXMLNodePreserveCDATA', 'NSXMLNodePreserveCharacterReferences', 'NSXMLNodePreserveDTD', 'NSXMLNodePreserveEmptyElements', 'NSXMLNodePreserveEntities', 'NSXMLNodePreserveNamespaceOrder', 'NSXMLNodePreservePrefixes', 'NSXMLNodePreserveQuotes',
```
Comment (justvanrossum) [inline Documentation/source/conf.py:239]: Not sure what this file does, but I' pretty sure the change here is not correct. At best, the "new" constants should be added here (in alphabetical place), but the old ones should probably stay.

File: Lib/vanilla/vanillaTextBox.py [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 _textAlignmentMap = {
-    "left": NSLeftTextAlignment,
-    "right": NSRightTextAlignment,
-    "center": NSCenterTextAlignment,
-    "justified": NSJustifiedTextAlignment,
+    "left": NSTextAlignmentLeft,
+    "right": NSTextAlignmentRight,
+    "center": NSTextAlignmentCenter,
+    "justified": NSTextAlignmentJustified,
     "natural": NSNaturalTextAlignment,
```
Comment (justvanrossum) [inline Lib/vanilla/vanillaTextBox.py:10]: Any reason to skip `NSNaturalTextAlignment` => `NSTextAlignmentNatural`?

File: Documentation/source/conf.py (hunk 1 above)
Comment (typemytype) [reply Documentation/source/conf.py:239]: this makes dummy classes and objects for the sphinx doc to work on other an OS

File: Lib/vanilla/vanillaTextBox.py (hunk 2 above)
Comment (schriftgestalt) [reply Lib/vanilla/vanillaTextBox.py]: no. Added it.

**Pull request merged**